### PR TITLE
[EC-183] Added RLP enc/dec for PrePrepare, Commit and NewRound messages

### PR DESCRIPTION
**Pull request by saltiniroberto**
## PR description
This PR adds the RLP encoding/decoding for the iBFT 2.0 PrePrepare, Commit and NewRound messages.

Tests will be added as part of another PR.

Also, see #153 for comments to the same PR raised before making the repo public.

### CODE REVIEW

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedCommitMessageData.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import tech.pegasys.pantheon.consensus.ibft.ConsensusRoundIdentifier;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessage.IbftV2;
+import tech.pegasys.pantheon.crypto.SECP256K1.Signature;
+import tech.pegasys.pantheon.ethereum.core.Hash;
+import tech.pegasys.pantheon.ethereum.rlp.RLPInput;
+import tech.pegasys.pantheon.ethereum.rlp.RLPOutput;
+
+public class IbftUnsignedCommitMessageData extends AbstractIbftUnsignedInRoundMessageData {
+  private static final int TYPE = IbftV2.COMMIT;
```
Comment (rain-on) [inline consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedCommitMessageData.java]: Having this information at this level is a smell - i.e. it appears in the Message object, and also at the underlying payload ... is it required to create signatures?

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedNewRoundMessageData.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  public IbftRoundChangeCertificate getRoundChangeCertificate() {
+    return roundChangeCertificate;
+  }
+
+  @Override
+  public void writeTo(final RLPOutput rlpOutput) {
+    // RLP encode of the message data content (round identifier and prepared certificate)
+    BytesValueRLPOutput ibftMessage = new BytesValueRLPOutput();
```
Comment (rain-on) [inline consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedNewRoundMessageData.java]: nit: is this an ibftMessage? Or is it the msgBuffer?

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessage/IbftCommitMessage.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+public class IbftCommitMessage extends AbstractIbftMessage {
+
+  private static final int MESSAGE_CODE = IbftV2.COMMIT;
+
+  private IbftCommitMessage(final ByteBuf data) {
+    super(data);
+  }
+
+  public static IbftCommitMessage fromMessage(final MessageData message) {
+    if (message instanceof IbftCommitMessage) {
```
Comment (jframe) [inline consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessage/IbftCommitMessage.java]: This logic is duplicated in each Ibft message, be nice to somehow move most of the common logic into a helper function in the abstract class.

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedCommitMessageData.java (hunk 1 above)
Comment (saltiniroberto) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedCommitMessageData.java]: Yes, it is required to create the signature.

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedNewRoundMessageData.java (hunk 2 above)
Comment (saltiniroberto) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedNewRoundMessageData.java]: This is a bug!

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessage/IbftCommitMessage.java (hunk 3 above)
Comment (saltiniroberto) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessage/IbftCommitMessage.java]: I'll see what I can do

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedNewRoundMessageData.java (hunk 2 above)
Comment (saltiniroberto) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedNewRoundMessageData.java]: Fixed

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessage/IbftCommitMessage.java (hunk 3 above)
Comment (saltiniroberto) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessage/IbftCommitMessage.java]: Done as discussed

**Pull request merged**